src/NLog.Loki/HttpLokiTransport.cs

```diff
@@ -50,7 +50,7 @@ private static async ValueTask ValidateHttpResponse(HttpResponseMessage response
 
             InternalLogger.Error("Failed pushing logs to Loki. Code: {Code}. Reason: {Reason}. Message: {Message}.",
                 response.StatusCode, response.ReasonPhrase, content);
-            throw new HttpRequestException("Failed pushing logs to Loki.", inner: null, response.StatusCode);
+            throw new HttpRequestException("Failed pushing logs to Loki.");
```

Comment:
Maybe change to [HttpException](https://docs.microsoft.com/en-us/dotnet/api/system.web.httpexception) like this `throw new HttpException((int)response.StatusCode, "Failed pushing logs to Loki.");`. The http-response-code can be extracted from the exception with [HttpException.GetHttpCode()](https://docs.microsoft.com/en-us/dotnet/api/system.web.httpexception.gethttpcode)

Reply:
And change `ThrowOnNonSuccessResponseCode` to `Assert.ThrowsAsync<HttpException>`

Reply:
Thank you. Your help is very much appreciated!
[HttpException](https://docs.microsoft.com/en-us/dotnet/api/system.web.httpexception?view=netframework-4.8#applies-to) is not supported by netstandard2.0 or net6.0. So I kept `HttpRequestException` with a null status code.

There's no reference to `HttpClient.ThrowOnNonSuccessResponseCode()`, since I do some custom validation in order to log the response.